invisible_cities/reco/corrections_new_test.py

```diff
@@ -183,23 +170,89 @@ def test_time_coefs_corr(map_filename, time):
+def test_get_df_to_z_converter_dv_known(map_filename, z):
+    z           = np.array(z)
+    map_z       = read_maps(map_filename)
+    z_converter = get_df_to_z_converter(map_z)
+    dv_mean     = np.mean(map_z.t_evol.dv)
+    assert_array_equal(z_converter(z), z * dv_mean)
+
+def test_get_df_to_z_converter_raises_exception_when_no_map(map_filename):
+    map_e = read_maps(map_filename)
+    assert_raises(TimeEvolutionTableMissing,
+                  get_df_to_z_converter,
+                  None)
+
+def test_get_df_to_z_converter_raises_exception_when_invalid_map(map_filename_MC):
+    map_e = read_maps(map_filename_MC)
+    assert_raises(TimeEvolutionTableMissing,
+                  get_df_to_z_converter,
+                  map_e)
+
+def test_get_normalization_factor_max_norm(correction_map_filename):
+    map_e  = read_maps(correction_map_filename)
+    factor = get_normalization_factor(map_e, norm_strategy.max)
+    norm   = map_e.e0.max().max()
+    assert factor == norm
+
+def test_get_normalization_factor_mean_norm(correction_map_filename):
+    map_e  = read_maps(correction_map_filename)
+    factor = get_normalization_factor(map_e, norm_strategy.mean)
+    norm   = np.mean(np.mean(map_e.e0))
+    assert factor == norm
+
+@given(floats(min_value = 1,
+              max_value = 1e4))
+def test_get_normalization_factor_custom_norm(correction_map_filename, custom_vaule):
+    map_e  = read_maps(correction_map_filename)
+    factor = get_normalization_factor(map_e,
+                                      norm_strategy.custom,
+                                      custom_vaule)
+    norm   = custom_vaule
+    assert factor == norm
+
+def test_get_normalization_factor_krscale(correction_map_filename):
+    map_e     = read_maps(correction_map_filename)
+    factor = get_normalization_factor(map_e, norm_strategy.kr)
+    kr_energy = 41.5575 * units.keV
+    assert factor == kr_energy
+
+def test_get_normalization_factor_custom_norm_raises_exception_when_no_value(map_filename):
+    map_e = read_maps(map_filename)
+    assert_raises(ValueError,
+                  get_normalization_factor,
+                  map_e, norm_strategy.custom)
+
+def test_get_normalization_factor_raises_exception_when_no_valid_norm(map_filename):
+    map_e = read_maps(map_filename)
+    assert_raises(ValueError,
+                  get_normalization_factor,
+                  map_e, None)
+
+def test__apply_all_correction_single_maps_raises_exception_when_no_map(map_filename):
+    map_e = read_maps(map_filename)
+    assert_raises(TimeEvolutionTableMissing,
+                  apply_all_correction_single_maps,
+                  map_e, map_e, None)
+
+def test_apply_all_correction_single_maps_raises_exception_when_invalid_map(map_filename_MC):
+    map_e = read_maps(map_filename_MC)
+    assert_raises(TimeEvolutionTableMissing,
                   apply_all_correction_single_maps,
-                  maps.e0, maps.lt, None, True)
+                  map_e, map_e, map_e)
```

Comment:
Add apply_temp = True so that it is obvious why the test fails (ie now it looks like it will always fail with MC map)